lib/util.js

```diff
@@ -317,7 +317,9 @@ Object.defineProperty(inspect, 'defaultOptions', {
     if (options === null || typeof options !== 'object') {
       throw new errors.TypeError('ERR_INVALID_ARG_TYPE', 'options', 'Object');
     }
-    Object.assign(inspectDefaultOptions, options);
```

Comment:
Any reason for not using `_extend`?

Reply:
The type check has to be done again in that case.

Reply:
I doubt it will add a noticeable overhead but yes it makes sense.

Reply:
1. Do we actually care about the performance of inspection? I thought we didn’t.
2. Same concern here as @lpinca. This is quite literally reinventing the wheel, and type checks for object are quite fast in modern V8.

Reply:
Updated the use _extend as it will still be faster than Object.assign

Reply:
About the performance - just recently I heard that `inspect` came up in a flamegraph. Seems like people use it a lot. I am aware that setting the default options will not be a big thing but the first implementation (not changing inspect for console.*) used it a lot and that is why I decided to optimize it as well.